target/linux/ar71xx/image/tp-link.mk

```diff
@@ -880,6 +880,21 @@ define Device/tl-wr1043nd-v4
 endef
 TARGET_DEVICES += tl-wr1043nd-v1 tl-wr1043nd-v2 tl-wr1043nd-v3 tl-wr1043nd-v4
 
+define Device/tl-wr942n-v1
+  DEVICE_TITLE := TP-LINK TL-WR942N v1
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
+  BOARDNAME := TL-WR942N-V1
+  TPLINK_BOARD_NAME := TL-WR942N
+  DEVICE_PROFILE := TLWR942NV1
+  IMAGE_SIZE := 14464k
+  KERNEL := kernel-bin | patch-cmdline | lzma | uImageArcher lzma
+  IMAGES := sysupgrade.bin factory.bin
+  IMAGE/sysupgrade.bin := append-rootfs | tplink-safeloader sysupgrade
+  IMAGE/factory.bin := append-rootfs | tplink-safeloader factory
+  MTDPARTS := spi0.0:128k(u-boot)ro,1344k(kernel),13120k(rootfs),64k(product-info)ro,64k(partition-table)ro,64k(soft-version)ro,64k(support-list)ro,640k(config)ro,576k(certificate)ro,64k(usb-config)ro,128k(log)ro,64k(radio-bk)ro,64k(ART)ro,14464k@0x20000(firmware)
```

Comment:
> I think we should have at least five last partitions rw as they are always updated during oem updates to have posibility to revert back to any oem firmware, leave partition product-info ro (it contains MAC and PIN) and combine all space between default-config and ART into one another ro partition.

NAK, please keep all vendor specific data partitions marked as _read-only_, as it's done for _all_ other similar (using tplink-safeloader) boards.

If, for any reasons, reverting back from LEDE to vendor firmware requires using combination of kernel+rootfs partitions from the same version of the vendor firmware as it was on device before upgrade to LEDE, make it clear in commit message and/or in the Wiki.